presto-tests/src/test/java/com/facebook/presto/tests/TestIsNotDistinctFrom.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+import com.facebook.presto.SystemSessionProperties;
+import com.facebook.presto.testing.LocalQueryRunner;
+import com.facebook.presto.testing.MaterializedResult;
+import com.facebook.presto.testing.QueryRunner;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class TestIsNotDistinctFrom
+{
+    @Test
+    public void testDoubleNan()
+    {
+        try (QueryRunner queryRunner = createQueryRunner()) {
+            MaterializedResult result = queryRunner.execute("select nan() is not distinct from nan()");
+            assertTrue((Boolean) result.getOnlyValue());
+        }
+    }
+
+    @Test
+    public void testRealNan()
+    {
+        try (QueryRunner queryRunner = createQueryRunner()) {
+            MaterializedResult result = queryRunner.execute("select cast(nan() as real) is not distinct from cast(nan() as real)");
+            assertTrue((Boolean) result.getOnlyValue());
+        }
+    }
+
+    @Test
+    public void testDistinctNan()
+    {
+        try (QueryRunner queryRunner = createQueryRunner()) {
+            MaterializedResult result = queryRunner.execute("select distinct a/a FROM (VALUES (0.0), (0.0)) x (a)");
```

Comment:
I don't see what test is trying to do

Reply:
This test was from the issue #1508 - since a/a (if a is double) is a nan there should only be one row.